File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/common/PrefUtils.java
package com.stringee.apptoappcallsample.common;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;

public class PrefUtils {
    private static SharedPreferences preferences;
    private static volatile PrefUtils instance;

    public PrefUtils(Context context) {
        preferences = context.getSharedPreferences(Constant.PREF_BASE, Context.MODE_PRIVATE);
    }

    public static synchronized PrefUtils getInstance(Context context) {
        if (instance == null) {
            synchronized (PrefUtils.class) {
                if (instance == null) {
                    instance = new PrefUtils(context);
                }
            }
        }
        return instance;
    }

    public int getInt(String key, int defValue) {
        return preferences.getInt(key, defValue);
    }

    public String getString(String key, String defValue) {
        return preferences.getString(key, defValue);
    }

    public boolean getBoolean(String key, boolean defValue) {
        return preferences.getBoolean(key, defValue);
    }

    public long getLong(String key, long defValue) {
        return preferences.getLong(key, defValue);
    }

    public void putInt(String key, int defValue) {
        Editor editor = preferences.edit();
        editor.putInt(key, defValue);
        editor.apply();
    }

    public void putString(String key, String defValue) {
        Editor editor = preferences.edit();
        editor.putString(key, defValue);
        editor.apply();
    }

    public void putBoolean(String key, boolean defValue) {
        Editor editor = preferences.edit();
        editor.putBoolean(key, defValue);
        editor.apply();
    }

    public void putLong(String key, long defValue) {
        Editor editor = preferences.edit();
        editor.putLong(key, defValue);
        editor.apply();
    }

    public void clearData() {
        Editor editor = preferences.edit();
        editor.clear();
        editor.apply();
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/common/PrefUtils.kt
package com.stringee.kotlin_onetoonecallsample.common

import android.content.Context
import android.content.SharedPreferences


class PrefUtils private constructor(applicationContext: Context) {
    init {
        preferences = applicationContext.getSharedPreferences(Constant.PREF_BASE, Context.MODE_PRIVATE)
    }

    fun getInt(key: String?, defValue: Int): Int {
        return preferences.getInt(key, defValue)
    }

    fun getString(key: String?, defValue: String?): String? {
        return preferences.getString(key, defValue)
    }

    fun getBoolean(key: String?, defValue: Boolean): Boolean {
        return preferences.getBoolean(key, defValue)
    }

    fun getLong(key: String?, defValue: Long): Long {
        return preferences.getLong(key, defValue)
    }

    fun putInt(key: String?, defValue: Int) {
        val editor = preferences.edit()
        editor.putInt(key, defValue)
        editor.apply()
    }

    fun putString(key: String?, defValue: String?) {
        val editor = preferences.edit()
        editor.putString(key, defValue)
        editor.apply()
    }

    fun putBoolean(key: String?, defValue: Boolean) {
        val editor = preferences.edit()
        editor.putBoolean(key, defValue)
        editor.apply()
    }

    fun putLong(key: String?, defValue: Long) {
        val editor = preferences.edit()
        editor.putLong(key, defValue)
        editor.apply()
    }

    fun clearData() {
        val editor = preferences.edit()
        editor.clear()
        editor.apply()
    }

    companion object {
        private lateinit var preferences: SharedPreferences

        @Volatile
        private var instance: PrefUtils? = null
        @Synchronized
        fun getInstance(context: Context): PrefUtils {
            return instance ?: synchronized(this) {
                instance
                    ?: PrefUtils(context.applicationContext).also {
                        instance = it
                    }
            }
        }
    }
}

